gqlalchemy/memgraph.py

```diff
@@ -585,3 +585,50 @@ def get_lambda(expression: str) -> str:
             return ""
         else:
             return f"(e, v | {expression})"
+
+
+class BreadthFirstSearch(IntegratedAlgorithm):
+    """Build a BFS call for a Cypher query.
+
+    The Breadth-First Search can be called in Memgraph with Cypher queries
+    such as:
+    MATCH (a {id: 723})-[*BFS ..10 (e, v | e.x > 12 AND v.y < 3)]-() RETURN *;
+    It is called inside the relationship clause, "* BFS" naming the algorithm,
+    "..10" specifying depth bounds, and "(e, v | <expression>)" is a filter
+    lambda.
+    """
+
+    def __init__(
+        self,
+        lower_bound: int = None,
+        upper_bound: int = None,
+        condition: str = None,
+    ) -> None:
+        super().__init__()
+        self.lower_bound = lower_bound
+        self.upper_bound = upper_bound
+        self.condition = condition
+
+    def __str__(self) -> str:
+        """get Cypher query string for this algorithm."""
+        algo_str = " * BFS"
+
+        bounds = self.get_bounds_str()
+        if bounds != "":
+            algo_str += f" {bounds}"
+
+        filter_lambda = super().get_lambda(self.condition)
+        if filter_lambda != "":
+            algo_str += f" {filter_lambda}"
+
+        return algo_str
+
+    def get_bounds_str(self) -> str:
```

Comment:
I think it would be better if you name this method as `to_cypher_bounds()`, since we already have `to_cypher_value()`, `to_cypher_properties()`, etc.